test/assertion_helper.rb

```diff
@@ -70,10 +69,10 @@ def inspect_source(source, cop, file = nil)
   end
 
   def investigate(cop, processed_source)
-    forces = RuboCop::Cop::Force.all.each_with_object([]) do |klass, instances|
-      next unless cop.join_force?(klass)
-
-      instances << klass.new([cop])
+    needed = Hash.new { |h, k| h[k] = [] }
```

Comment:
I'm... not sure why this is needed to be defined like this (v.s. `needed = Hash.new`) 🤔

Reply:
Good question! It's another Ruby nicety

This creates a hash where the default value is a new hash (new as in new every time, not the same array as it'd happen with `Hash.new([])`)
This allows to use `<<` in the line below without having to initialize each `needed[x]` every time.

More details:
- https://endofline.wordpress.com/2010/12/24/hash-tricks/
- https://netinlet.com/blog/2014/01/01/fun-with-ruby-hashes/